core/include/keyman/keyman_core_api.h

```diff
@@ -526,6 +558,152 @@ enum km_core_action_type {
   KM_CORE_IT_MAX_TYPE_ID
 };
 
+/*
+```
+### Actions
+This structure provides the results of processing a key event to the Platform layer and
+should be processed by the Platform layer to issue commands to the os text
+services framework to transform the text store in the Client Application, among
+other actions.
+
+This API replaces the Action items APIs, which is now deprecated and will be
+removed in the future.
+```c
+*/
+
+typedef enum { KM_CORE_FALSE = 0, KM_CORE_TRUE = 1 } km_core_bool;
+typedef enum { KM_CORE_CAPS_UNCHANGED = -1, KM_CORE_CAPS_OFF = 0, KM_CORE_CAPS_ON = 1 } km_core_caps_state;
+
+typedef struct {
+  // number of codepoints (not codeunits!) to delete from app context.
+  unsigned int code_points_to_delete;
+
+  // null-term string of characters to insert into document
+  km_core_usv* output;
+
+  // list of options to persist, terminated with KM_CORE_OPTIONS_END
+  km_core_option_item* persist_options;
+
+  // issue a beep, 0 = no, 1 = yes
+  km_core_bool do_alert;
+
+  // emit the input keystroke to the application, unmodified? 0 = no, 1 = yes
```

Comment:
unmodified just means 'no modification'. We don't support modification of the keystroke.

Reply:
Ok, now I understand what's meant. The `unmodified` refers mainly to `input keystroke`, not `no/yes`. And it's not a left-over question from API design. 

This might be easier to understand if the word order is different or the question mark omitted:

```suggestion
  // emit the unmodified input keystroke to the application? 0 = no, 1 = yes
```

or

```suggestion
  // emit the (unmodified) input keystroke to the application? 0 = no, 1 = yes
```

or

```suggestion
  // emit the input keystroke to the application, unmodified. 0 = no, 1 = yes
```